packages/synapse-react-client/src/components/AclEditor/AclEditor.tsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react'
+import { Box, Collapse, Typography } from '@mui/material'
+import { ResourceAccessItem } from './ResourceAccessItem'
+import UserSearchBoxV2 from '../UserSearchBox/UserSearchBoxV2'
+import { ResourceAccess } from '@sage-bionetworks/synapse-types'
+import useUpdateAcl from './useUpdateAcl'
+import { TransitionGroup } from 'react-transition-group'
+import { PermissionLevel } from '../../utils/PermissionLevelToAccessType'
+import { SynapseSpinner } from '../LoadingScreen/LoadingScreen'
+
+export type AclEditorProps = {
+  isLoading: boolean
+  resourceAccessList: ResourceAccess[]
+  availablePermissionLevels: PermissionLevel[]
+  emptyText: React.ReactNode
+  addResourceAccessItem: ReturnType<
+    typeof useUpdateAcl
+  >['addResourceAccessItem']
+  updateResourceAccessItem: ReturnType<
+    typeof useUpdateAcl
+  >['updateResourceAccessItem']
+  removeResourceAccessItem: ReturnType<
+    typeof useUpdateAcl
+  >['removeResourceAccessItem']
+}
+
+export function AclEditor(props: AclEditorProps) {
+  const {
+    isLoading,
+    resourceAccessList,
+    availablePermissionLevels,
+    emptyText,
+    addResourceAccessItem,
+    updateResourceAccessItem,
+    removeResourceAccessItem,
+  } = props
+  if (isLoading) {
+    return (
+      <Box display={'flex'} justifyContent={'center'} height={'150px'}>
```

Comment:
One stylistic question - I've been using sx styles to change these, which I've found really convenient for rapid iterative development.  Is there a reason to use the specific component properties (like `display`), or is this just a preference?

Reply:
I tend to use the props if they are there, [which is true for Box, Stack, Typography, and Grid](https://mui.com/system/getting-started/usage/#api-tradeoff)

For cases where we would want to share the same set of styles across multiple `Box` instances, we'd probably be better of using a shared `sx` object or styled component. For one-offs, I don't really have a preference. The props feel more natural to me.